src/modules/caen/CaenBase.cpp

```diff
@@ -213,21 +221,34 @@ void CaenBase::processingThread() {
       usleep(10);
     }
 
-    if (ProduceTimer.timeout()) {
+    if (ProduceTimer.timeout()){
       // XTRACE(DATA, DEB, "Serializer timer timed out, producing message now");
       RuntimeStatusMask =
           RtStat.getRuntimeStatusMask({ITCounters.RxPackets, Counters.Events,
                                        Counters.KafkaStats.produce_bytes_ok});
+      // Produce messages if there are events
+      for (auto &Serializer: Serializers) {
+        Serializer->produce();
+      }
+      MonitorSerializer->produce();
```

Comment:
I questioned previously if this produce was necessary. I now think that it _should_ be removed barring changes in `AR51Serializer`

Only the `serialize` method clears the internal buffer of `MonitorSerialzer` , so any call to `produce` always has data (if `DataLen` was not 0 above). Every time the timer times-out, this `produce` call will re-send the last serialized readout sample data.
If the detector sends no further events, the last sampled data will be sent to Kafka once per second as long as the EFU is running.